core/src/impl/Kokkos_AnalyzePolicy.hpp

```diff
@@ -178,11 +202,18 @@ struct AnalyzePolicy<Base>
                                                      , void // TODO set default iteration pattern
                                                      , typename Base::iteration_pattern
                                                      >::type;
+
+  using launch_bounds = typename std::conditional< is_void< typename Base::launch_bounds >::value
+                                                     , LaunchBounds<1024,1> // TODO set correct bounds for each device
```

Comment:
The default launch bounds are set to be maximally permissive. You can't have more than 1024 threads in a block or less than 1 block per SM. 

But yours is a good question for the future. It's possible that future architectures allow more threads per block.